test/python/test_ibmqjob_states.py

```diff
@@ -211,9 +213,8 @@ def test_block_on_result_waiting_until_exception(self):
         with ThreadPoolExecutor() as executor:
             executor.submit(_auto_progress_api, self._current_api)
 
-        result = job.result()
-        self.assertEqual(result.get_status(), 'ERROR')
-        self.assertStatus(job, JobStatus.ERROR)
+        with self.assertRaises(JobError):
```

Comment:
An Exception should not change the Job state. If result() raises is due to a Timeout, or due to something wrong happened while requesting the status/job info to the server, we don't really know in which state the Job could be. Is server responsibility to give us this state, because it could be in any state (except INITIALIZING).

Reply:
Ok, let's leave this test as it is now and add another one testing that if there is a non-job-related error, the state of the job does not change.

Reply:
I think I'm missing it but where is the test that checks the state of a job does not progress if there is a non-job-related error?

Reply:
Ok, too many things to review I guess, but I thought that we were not finally doing this test, because it's kind of pointless/artificial, as a Job can actually progress if we are getting a non-job-related error, we just don't know.

Reply:
As a reminder to myself, there is no way of getting an out-of-date status except it raises, in which case we don't get a new status at all. It makes sense to omit this test.